### PR TITLE
core: update _rm_titles to account for title argument name bug

### PR DESCRIPTION
**Pull request by rsk2327**
Issue : For functions which have an argument with the name 'title', the convert_pydantic_to_openai_function generates an incorrect output and omits the argument all together. This is because the _rm_titles function removes all instances of the the key 'title' from the output.

Description : Updates the _rm_titles function to check the presence of the 'type' key as well before removing the 'title' key. As the title key that we wish to omit always has a type key along with it. 

Potential gap if there is a function defined which has both title and key as argument names, in which case this would fail. Maybe we could set a filter on the function argument names and reject those with keyword argument names.

No dependencies. Passed all tests. 

- [x] **PR title**: "package: description"
  - Where "package" is whichever of langchain, community, core, experimental, etc. is being modified. Use "docs: ..." for purely docs changes, "templates: ..." for template changes, "infra: ..." for CI changes.
  - Example: "community: add foobar LLM"

- [x] **PR message**: ***Delete this entire checklist*** and replace with
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change
    - **Twitter handle:** if your PR gets announced, and you'd like a mention, we'll gladly shout you out!

- [x] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, hwchase17.

### CODE REVIEW

File: libs/core/langchain_core/utils/function_calling.py [hunk 1]
```diff
@@ -54,7 +54,8 @@ class ToolDescription(TypedDict):
 def _rm_titles(kv: dict) -> dict:
     new_kv = {}
     for k, v in kv.items():
-        if k == "title":
+        # print(kv.keys())
```
Comment (eyurtsev) [inline libs/core/langchain_core/utils/function_calling.py:57]:
```suggestion
```

**Comment by rsk2327** [comment]
@baskaryan Let me know if it makes more sense to set an explicit check on function argument names to raise an error for argument names like 'title' and 'key'. Or if the above solution is good enough

**Comment by rsk2327** [comment]
@baskaryan @eyurtsev Let me know if any changes are required to move forward.

File: libs/core/langchain_core/utils/function_calling.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ class ToolDescription(TypedDict):
 def _rm_titles(kv: dict) -> dict:
     new_kv = {}
     for k, v in kv.items():
-        if k == "title":
+        if k == "title" and "type" in kv.keys():
```
Comment (baskaryan) [inline libs/core/langchain_core/utils/function_calling.py:57]: could we add a unit test for this?

**Comment by rsk2327** [comment]
@baskaryan Added unit tests. 

Also updated the logic for the function to account for the edge case that I had previously mentioned. The new logic should hold irrespective of the use of keywords like 'title', 'type' and 'properties' as Class variable names

**Comment by rsk2327** [comment]
@baskaryan @eyurtsev

**Review by hwchase17 — APPROVED** [review]
